controllers/cinder_controller.go

```diff
@@ -341,16 +344,41 @@ func (r *CinderReconciler) reconcileInit(
 func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinderv1beta1.Cinder, helper *helper.Helper) (ctrl.Result, error) {
 	r.Log.Info(fmt.Sprintf("Reconciling Service '%s'", instance.Name))
 
-	// If the service object doesn't have our finalizer, add it.
-	controllerutil.AddFinalizer(instance, helper.GetFinalizer())
-	// Register the finalizer immediately to avoid orphaning resources on delete
-	if err := r.Update(ctx, instance); err != nil {
+	if !controllerutil.ContainsFinalizer(instance, helper.GetFinalizer()) {
+		// If the service object doesn't have our finalizer, add it.
+		controllerutil.AddFinalizer(instance, helper.GetFinalizer())
+		// Register the finalizer immediately to avoid orphaning resources on delete
+		err := r.Update(ctx, instance)
+
 		return ctrl.Result{}, err
 	}
 
 	// ConfigMap
 	configMapVars := make(map[string]env.Setter)
 
+	//
+	// create RabbitMQ transportUrl CR and get the actual URL from the associated secret that is created
+	//
+
+	transportUrl, op, err := r.transportUrlCreateOrUpdate(instance)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("TransportURL %s successfully reconciled - operation: %s", transportUrl.Name, string(op)))
+	}
+
+	instance.Status.TransportUrlSecret = transportUrl.Status.SecretName
+
+	if instance.Status.TransportUrlSecret == "" {
+		r.Log.Info(fmt.Sprintf("Waiting for TransportURL %s secret to be created", transportUrl.Name))
+		return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+	}
+
```

Comment:
A new condition might be warranted, but here in the `Cinder` controller it does sort of make sense to bundle this operation under "preparing the inputs", so that is why I haven't added another condition (yet).  As for the secret hash, the secret itself is hashed where it is actually used (here it is just being indirectly created).  Controllers that actually use the secret do the hashing [1].

[1] https://github.com/abays/cinder-operator/blob/16cc41729591f406a8400a9bd494d6ab55023268/controllers/cinderbackup_controller.go#L302

Reply:
I see this as an equivalent to the `DBReadyCondition` because we are waiting for RabbitMQ to be ready and if the rabbit operator is having trouble deploying it will be helpful to easily know where the cinder-operator is stuck.

I believe we need to change old line 481 (new line 509) where we say:
```
// TODO: These will not work without rabbit yet
```
To either remove it, or just say that at that point Rabbit is already running.

Reply:
Sure, that makes sense.  I will add a condition for the transport instead of rolling it into the input-ready condition.